src/engine/objects/mod.rs

```diff
@@ -586,8 +612,10 @@ where
     }
 }
 
+/// Represents a reference to a [`Node`] object as either an [`Identifier`]
+/// containing a type and id, or a complete [`Node`] struct.
 #[derive(Clone, Debug)]
-pub(crate) enum NodeRef<GlobalCtx: GlobalContext, RequestCtx: RequestContext> {
+pub enum NodeRef<GlobalCtx: GlobalContext, RequestCtx: RequestContext> {
```

Comment:
Do custom resolvers need to create these directly, as opposed to going through the ResolverFacade?

Reply:
I believe so, likely also an oversight.

Reply:
Fixed.